### PR TITLE
Azure RM Templates for the Azure.DevTestLab resource provider.

### PR DESCRIPTION
**Pull request by mithunshanbhag**
Azure RM Templates for the Azure.DevTestLab resource provider.

We have made available the following RM templates:
- Create a new lab.
- Create a new VM in a lab using a built-in account.
- Create a new VM in a lab using a username/password combination.
- Create a new VM in a lab using a username/SSHKey combination.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@mithunshanbhag__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (mithuns). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by singhkay** [comment]
@template-deploy-bot test this please

File: 101-devtestlab-create-vm-builtin-user/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "vmName": {
+      "type": "string",
+      "metadata": {
+        "description": "The name of the new vm to be created."
+      }
+    },
+    "labName": {
```
Comment (singhkay) [inline 101-devtestlab-create-vm-builtin-user/azuredeploy.json]: prefix with existing to set the right expectations for this parameter

File: 101-devtestlab-create-vm-builtin-user/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      "type": "string",
+      "metadata": {
+        "description": "The name of an existing lab where the new vm will be created. The lab's resource group must be the same as the one in which this template is being deployed."
+      }
+    },
+    "vmTemplateName": {
```
Comment (singhkay) [inline 101-devtestlab-create-vm-builtin-user/azuredeploy.json]: prefix with existing to set the right expectations for this parameter

File: 101-devtestlab-create-vm-builtin-user/metadata.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "itemDisplayName": "Create a new VM in a DevTestLab instance",
```
Comment (singhkay) [inline 101-devtestlab-create-vm-builtin-user/metadata.json]: These values look the same for the first template above, needs to be different for this template

File: 101-devtestlab-create-vm-username-pwd/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "vmName": {
+      "type": "string",
+      "metadata": {
+        "description": "The name of the new vm to be created."
+      }
+    },
+    "labName": {
```
Comment (singhkay) [inline 101-devtestlab-create-vm-username-pwd/azuredeploy.json]: prefix with existing to set the right expectations for this parameter

File: 101-devtestlab-create-vm-username-pwd/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+      "type": "string",
+      "metadata": {
+        "description": "The name of an existing lab where the new vm will be created. The lab's resource group must be the same as the one in which this template is being deployed."
+      }
+    },
+    "vmTemplateName": {
```
Comment (singhkay) [inline 101-devtestlab-create-vm-username-pwd/azuredeploy.json]: prefix with existing to set the right expectations for this parameter

File: 101-devtestlab-create-vm-username-pwd/metadata.json [hunk 6]
```diff
@@ -0,0 +1,7 @@
+{
+  "itemDisplayName": "Create a new VM in a DevTestLab instance",
```
Comment (singhkay) [inline 101-devtestlab-create-vm-username-pwd/metadata.json]: Needs to be different for this template

File: 101-devtestlab-create-vm-username-ssh/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+  "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "vmName": {
+      "type": "string",
+      "metadata": {
+        "description": "The name of the new vm to be created."
+      }
+    },
+    "labName": {
```
Comment (singhkay) [inline 101-devtestlab-create-vm-username-ssh/azuredeploy.json]: prefix with existing to set the right expectations for this parameter

File: 101-devtestlab-create-vm-username-ssh/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+      "type": "string",
+      "metadata": {
+        "description": "The name of an existing lab where the new vm will be created. The lab's resource group must be the same as the one in which this template is being deployed."
+      }
+    },
+    "vmTemplateName": {
```
Comment (singhkay) [inline 101-devtestlab-create-vm-username-ssh/azuredeploy.json]: prefix with existing to set the right expectations for this parameter

File: 101-devtestlab-create-vm-username-ssh/metadata.json [hunk 9]
```diff
@@ -0,0 +1,7 @@
+{
+  "itemDisplayName": "Create a new VM in a DevTestLab instance",
```
Comment (singhkay) [inline 101-devtestlab-create-vm-username-ssh/metadata.json]: Needs to be different for this template

**Comment by singhkay** [comment]
lgtm, merging

**Pull request merged**